meta-balena-common/recipes-core/balena-rollback/files/rollback-altboot

```diff
@@ -23,7 +23,7 @@ if ! rollback-check-version; then
 	exit 1
 fi
 
-if [ "$resin_root_part" != "$current_part_idx" ]; then
+if [ "$resin_root_part" != "$current_part_idx" ] && [ "$resin_root_part" != "$current_part_name" ]; then
```

Comment:
what's the context to switch to names from indexes?

Reply:
I always thought we were using the numbers because of the limitations of the bootloaders. HUP & rollback use `/mnt/sysroot/active` and `/mnt/sysroot/inactive` for all the destructive actions and then need to figure out the partition number only to update the bootloader file. Also on most device types this is hardcoded to 2 and 3 anyway and we actually had to create a separate GRUB config hardcoding 3 and 4 for secure boot, because it shifts when boot and EFI partitions are split, so now we ship 2 GRUB config variants on generic-amd64.

Since the balena bootloader is written from scratch and linux can parse the labels, it felt like a good opportunity to clean it up right away. But I do not insist on doing it in one go, I can rework it to use the numbers and discuss this separately.

Reply:
> it felt like a good opportunity to clean it up right away

Thanks, I agree. Just needed the context :)